src/java/main/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -915,11 +915,13 @@ protected void pRequest(Request request) throws RequestProcessorException {
     private List<ACL> removeDuplicates(List<ACL> acl) {
 
         ArrayList<ACL> retval = new ArrayList<ACL>();
-        Iterator<ACL> it = acl.iterator();
-        while (it.hasNext()) {
-            ACL a = it.next();
-            if (retval.contains(a) == false) {
-                retval.add(a);
+        if(acl != null) {
```

Comment:
Server side, the validation for null is already present in fixupACL method and it throws InvalidACLException.  This check is after the removeDuplicates method call.

So do you mean we can handle null in removeDuplicates method itself and if so,, we can remove the null validation code in fixupACL method.